client/src/components/LinkButton.jsx

```diff
@@ -1,56 +1,53 @@
-import React, { useEffect, useState } from 'react';
+import React from 'react';
 import propTypes from 'prop-types';
-
-import { useLink } from '../services';
+import Button from 'plaid-threads/Button';
+import { usePlaidLink } from 'react-plaid-link';
 
 LinkButton.propTypes = {
-  userId: propTypes.number,
-  itemId: propTypes.number,
-  altClasses: propTypes.string,
-  primary: propTypes.bool,
+  linkToken: propTypes.string,
+  callbacks: propTypes.object,
+  update: propTypes.bool,
 };
 
 LinkButton.defaultProps = {
-  userId: null,
-  itemId: null,
-  altClasses: null,
-  primary: false,
+  linkToken: null,
+  callbacks: null,
+  update: false,
 };
 
-export default function LinkButton({
-  children,
-  userId,
-  itemId,
-  altClasses,
-  primary,
-}) {
-  const [linkInstance, setLinkInstance] = useState({});
-  const { linkHandlers, getLinkHandler } = useLink();
-
-  const isPrimary = primary ? 'button--is-primary' : '';
-  const classlist = altClasses !== null ? altClasses : '';
-
-  useEffect(() => {
-    getLinkHandler({ userId, itemId });
-  }, [userId, itemId, getLinkHandler]);
+export default function LinkButton({ children, linkToken, callbacks, update }) {
+  const linkConfig = {
+    ...callbacks,
+    token: linkToken,
+  };
 
-  useEffect(() => {
-    if (itemId) {
-      setLinkInstance(linkHandlers.byItem[itemId] || {});
-    } else {
-      setLinkInstance(linkHandlers.byUser[userId] || {});
-    }
-  }, [linkHandlers, itemId, userId]);
+  const { open, ready } = usePlaidLink(linkConfig);
 
   return (
-    <button
-      className={`button ${isPrimary} ${classlist}`}
-      disabled={!linkInstance.isReady}
-      onClick={() => {
-        linkInstance.handler.open();
-      }}
-    >
-      {children}
-    </button>
+    <>
+      {!update && (
+        <Button
+          centered
+          inline
+          disabled={!ready}
+          onClick={() => {
+            open();
+          }}
+        >
+          {children}
+        </Button>
+      )}
+      {update && ( // case where link is launched in update mode from dropdown menu in the
+        // item card after item is set to "bad state"
+        <div
+          disabled={!ready}
+          onClick={() => {
+            open();
+          }}
+        >
+          {children}
+        </div>
```

Comment:
divs should not have onclicks, and `disabled` does not do anything do a div. Should this be a `Button`?